components/psst/browser/core/rule_data_reader_unittest.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/psst/browser/core/rule_data_reader.h"
+
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "base/base_paths.h"
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/logging.h"
+#include "base/path_service.h"
+#include "brave/components/constants/brave_paths.h"
+#include "brave/components/psst/browser/core/matched_rule.h"
+#include "brave/components/psst/browser/core/psst_rule.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace psst {
+
+namespace {
+std::string ReadFile(const base::FilePath& file_path) {
+  std::string contents;
+  bool success = base::ReadFileToString(file_path, &contents);
+  if (!success || contents.empty()) {
+    VLOG(2) << "ReadFile: cannot " << "read file " << file_path;
+  }
+  return contents;
+}
+}  // namespace
+
+class RuleDataReaderUnitTest : public testing::Test {
+ public:
+  void SetUp() override {
+    base::FilePath test_data_dir(
+        base::PathService::CheckedGet(brave::DIR_TEST_DATA));
+    test_data_dir_base_ = test_data_dir.AppendASCII("psst-component-data");
+
+    auto psst_rules_content = ReadFile(test_data_dir_base_.Append(
+        base::FilePath::FromUTF8Unsafe("psst.json")));
```

Comment:
it's much better if we embed this in the file instead of reading it from psst.json because it makes the test harder to understand. Just include the json here directly